src/components/input/DateInput.vue

```diff
@@ -0,0 +1,34 @@
+<script>
+import { defineComponent } from 'vue';
+import BaseInput from '@/components/base/BaseInput.vue';
+
+export default defineComponent({
+  components: {
+    BaseInput,
+  },
+  data() {
+    return {
+      isTypeDate: false,
+    };
+  },
+  updated() {
+    if (this.$refs.dateInput.modelValue) this.isTypeDate = true;
+  },
```

Comment:
You could implement a nested v-model here and have this logic handled in the `@update:model-value` event handler (which would handle this + re-emit the event to the parent) 😄